stackslib/src/chainstate/stacks/block.rs

```diff
@@ -1699,6 +1973,135 @@ mod test {
+            TransactionSpendingCondition::new_multisig_order_independent_p2wsh(
+                2,
+                vec![pubk_1.clone(), pubk_2.clone(), pubk_3.clone()],
+            )
+            .unwrap();
+
+        let order_independent_multisig_condition_p2sh =
+            TransactionSpendingCondition::new_multisig_order_independent_p2sh(
+                2,
+                vec![pubk_1.clone(), pubk_2.clone(), pubk_3.clone()],
+            )
+            .unwrap();
+
+        let order_independent_sponsored_auth_p2sh = TransactionAuth::Sponsored(
+            TransactionSpendingCondition::new_singlesig_p2pkh(StacksPublicKey::from_private(
+                &privk,
+            ))
+            .unwrap(),
+            order_independent_multisig_condition_p2sh.clone(),
+        );
+
+        let order_independent_sponsored_auth_p2wsh = TransactionAuth::Sponsored(
+            TransactionSpendingCondition::new_singlesig_p2pkh(StacksPublicKey::from_private(
+                &privk,
+            ))
+            .unwrap(),
+            order_independent_multisig_condition_p2wsh.clone(),
+        );
+        let order_independent_origin_auth_p2sh =
+            TransactionAuth::Standard(order_independent_multisig_condition_p2sh.clone());
+
+        let order_independent_origin_auth_p2wsh =
+            TransactionAuth::Standard(order_independent_multisig_condition_p2wsh.clone());
+
+        let order_independent_multisig_tx_transfer_mainnet_p2sh = StacksTransaction::new(
+            TransactionVersion::Mainnet,
+            order_independent_origin_auth_p2sh.clone(),
+            TransactionPayload::TokenTransfer(
+                stx_address.into(),
+                123,
+                TokenTransferMemo([1u8; 34]),
+            ),
+        );
+
+        let order_independent_multisig_tx_transfer_mainnet_p2wsh = StacksTransaction::new(
+            TransactionVersion::Mainnet,
+            order_independent_origin_auth_p2wsh.clone(),
+            TransactionPayload::TokenTransfer(
+                stx_address.into(),
+                123,
+                TokenTransferMemo([1u8; 34]),
+            ),
+        );
+
+        let order_independent_sponsored_multisig_tx_transfer_mainnet_p2sh = StacksTransaction::new(
+            TransactionVersion::Mainnet,
+            order_independent_sponsored_auth_p2sh.clone(),
+            TransactionPayload::TokenTransfer(
+                stx_address.into(),
+                123,
+                TokenTransferMemo([1u8; 34]),
+            ),
+        );
+
+        let order_independent_sponsored_multisig_tx_transfer_mainnet_p2wsh = StacksTransaction::new(
+            TransactionVersion::Mainnet,
+            order_independent_sponsored_auth_p2wsh.clone(),
+            TransactionPayload::TokenTransfer(
+                stx_address.into(),
+                123,
+                TokenTransferMemo([1u8; 34]),
+            ),
+        );
+
+        let mut tx_signer =
```

Comment:
It looks like most if not all of our multisig tests are testing using 2-of-3. How about a little variety? 2-of-2, 1-of-4, etc?

Reply:
Never mind, there's a ton of those in `transaction.rs`